scripts/linters/pre_commit_linter.py

```diff
@@ -418,29 +419,52 @@ def categorize_files(file_paths):
     _FILES.update(all_filepaths_dict)
 
 
-def _print_complete_summary_of_errors(all_messages):
-    """Print complete summary of errors."""
-    error_messages = all_messages
-    if error_messages != '':
+def _print_complete_summary_of_lint_messages(lint_messages):
+    """Print complete summary of lint messages."""
+    if lint_messages != '':
         python_utils.PRINT('Summary of Errors:')
         python_utils.PRINT('----------------------------------------')
-        for message in error_messages:
+        for message in lint_messages:
             python_utils.PRINT(message)
 
 
-def _get_task_output(all_messages, task, semaphore):
+def _get_task_output(lint_messages, error_messages, task, semaphore):
     """Returns output of running tasks.
 
     Args:
-        all_messages: list(str). List of summary messages of linter output.
+        lint_messages: list(str). List of summary messages of linter output.
+        error_messages: list(str). List of error messages during linter
+            execution.
         task: object(TestingTaskSpec). The task object to get output of linter.
         semaphore: threading.Semaphore. The object that controls how many tasks
             can run at any time.
     """
-    all_messages += task.output
+    try:
+        lint_messages += task.output
+    except Exception:
```

Comment:
Why are we adding try-except here? I mean it's not executing the code right? (I think it's just appending the output of the task into the list, right?)

Reply:
The reason is if there is some error during a lint check execution then we have `None` as `task.output` and we can not concatenate a `None` to a list. So it will raise an exception and we know that an error has happened somewhere.

Reply:
In that case, we can have `if-else` check, instead of try-catch, right?

Reply:
Used if-else